bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -2014,24 +2033,33 @@ else if (i.alphaData != null) {
 			if (hIcon == 0) SWT.error(SWT.ERROR_NO_HANDLES);
 			OS.DeleteObject(hBitmap);
 			OS.DeleteObject(hMask);
-			image.handle = hIcon;
+			image.setHandleForZoomLevel(hIcon, i.height, i.width, zoomLevel);
 			image.type = SWT.ICON;
 		}
 	} else {
 		if (image == null) {
 			result = new long []{hDib};
 		} else {
-			image.handle = hDib;
+			image.setHandleForZoomLevel(hDib, i.height, i.width, zoomLevel);
 			image.type = SWT.BITMAP;
 			image.transparentPixel = i.transparentPixel;
 		}
 	}
 	return result;
 }
 
-static long [] init(Device device, Image image, ImageData source, ImageData mask) {
+private void setHandleForZoomLevel(long handle, int height, int width, Integer zoomLevel) {
+	if (this.handle == 0) {
+		this.handle = handle;	// Set handle for default zoom level
+	}
+	if (zoomLevel != null && handleMap.get(zoomLevel) == null) {
```

Comment:
Can't you just use handleMap.contains(zoomLevel) for the second check?

Reply:
containsKey, yes good idea